### PR TITLE
Removed BuddyType attribute and added ModelMetadatType 

### PR DESCRIPTION
**Pull request by kirthik**
Removed BuddyType attribute and added ModelMetadatType instead. Added AutoMapper for Desktop scenario. I am still using SimpleMapper for Core scenario.

@DamianEdwards, @yishaigalatzer - Could you please take a look?

### CODE REVIEW

File: src/MusicStore.Spa/Apis/AlbumsApiController.cs [hunk 1]
```diff
@@ -25,14 +28,20 @@ public AlbumsApiController(MusicStoreContext storeContext)
...
+            //.Include(a => a.Artist)
+#if ASPNET50
+            await albums.ToPagedListAsync(page, pageSize, sortBy,
+                    a => a.Title,                                    // sortExpression
+                    SortDirection.Ascending,                         // defaultSortDirection
+                    a => Mapper.Map(a, new AlbumResultDto()));       // selector
+#else
+            await albums.ToPagedListAsync(page, pageSize, sortBy,
                     a => a.Title,                                    // sortExpression
                     SortDirection.Ascending,                         // defaultSortDirection
```
Comment (yishaigalatzer) [inline src/MusicStore.Spa/Apis/AlbumsApiController.cs]: nit: you can just `ifdef` the relevant line

**Comment by yishaigalatzer** [comment]
looks much cleaner to me.

@DamianEdwards we also finished the action result pattern that we thought made sense, should we follow up with a second PR to remove ApiResult?

File: src/MusicStore.Spa/Apis/AlbumsApiController.cs [hunk 2]
```diff
@@ -99,7 +122,11 @@ public AlbumsApiController(MusicStoreContext storeContext)
 
             // Save the changes to the DB
             var dbAlbum = new Album();
+#if ASPNET50
+            await _storeContext.Albums.AddAsync(Mapper.Map(album, dbAlbum));
```
Comment (yishaigalatzer) [inline src/MusicStore.Spa/Apis/AlbumsApiController.cs]: Wrap the mappers into a single mapper abstraction

File: src/MusicStore.Spa/Infrastructure/SimpleMapper.cs [hunk 3]
```diff
@@ -14,8 +14,12 @@ public class SimpleMapper
 
         public static TDest Map<TSource, TDest>(TSource source, TDest dest)
         {
+#if ASPNET50
+            return AutoMapper.Mapper.Map(source, dest);
+#else
             var map = (Func<TSource, TDest, TDest>)_mapCache.GetOrAdd(Tuple.Create(typeof(TSource), typeof(TDest)), _ => MakeMapMethod<TSource, TDest>());
             return map(source, dest);
+#endif
```
Comment (kirthik) [inline src/MusicStore.Spa/Infrastructure/SimpleMapper.cs]: @yishaigalatzer - Moved AutoMapper mapping to SimpleMapper Map method. Does this look better?

Comment (yishaigalatzer) [inline src/MusicStore.Spa/Infrastructure/SimpleMapper.cs]: Much better, even better would be to wrap in a different interface 

-----Original Message-----

**Comment by rynowak** [comment]
@kirthik is there a reason this is still sitting here?

**Comment by kirthik** [comment]
Oops no, lost track of it. I had to get sign off from @DamianEdwards and @yishaigalatzer. I will get it in ASAP.

**Comment by rynowak** [comment]
awesome! this is going to unblock some of the stuff that I need to do here

**Comment by kirthik** [comment]
If code looks fine to you, could you please sign off?

**Comment by DamianEdwards** [comment]
@kirthik It seems you've deleted `SimpleMapper` completely? If so, what are we using in .NET Core scenario now?

**Comment by yishaigalatzer** [comment]
Automapper now supports coreclr

**Comment by DamianEdwards** [comment]
:shipit:

**Comment by kirthik** [comment]
cc @rynowak - This is in.

**Comment by rynowak** [comment]
Awesome thanks!

**Pull request closed without merging**